main.js

```diff
@@ -0,0 +1,312 @@
+  return keyboardContainer;
+};
+
+const generateKeys = (data) => {
+  const keys = [];
+  data.forEach((key) => {
+    keys.push(new Keys(key));
+  });
+  return keys;
+};
+
+const renderKeysToDom = () => {
+  arrayWithKeys.forEach((element) => {
+    const wrapper = getWrapper();
+    generateKeys(element).forEach((key) => {
+      wrapper.append(key.generateKey());
+    });
+  });
+};
+
+const setDefaultLanguage = () => {
+  if (lang === 'en') {
+    document.querySelectorAll('sup').forEach((element) => {
+      element.classList.add('lang_reset');
+    });
+  } else {
+    document.querySelectorAll('span').forEach((element) => {
+      element.classList.add('lang_reset');
+    });
+  }
+};
+
+// feat: CapsLock
+function upperLower() {
+  capsLock = !capsLock;
+  if (capsLock) {
+    document.querySelectorAll('span').forEach((element) => {
+      element.classList.add('uppercase');
+      element.classList.remove('lowercase');
+    });
+    document.querySelectorAll('sup').forEach((element) => {
+      element.classList.add('uppercase');
+      element.classList.remove('lowercase');
+    });
+  } else {
+    document.querySelectorAll('span').forEach((element) => {
+      element.classList.add('lowercase');
+      element.classList.remove('uppercase');
+    });
+    document.querySelectorAll('sup').forEach((element) => {
+      element.classList.add('lowercase');
+      element.classList.remove('uppercase');
+    });
+  }
+}
+
+function addTextAtCaret(text) {
+  const textArea = document.getElementById('result');
+  let cursorPosition = textArea.selectionStart;
+  let front;
+  let back;
+  const inputText = '';
+
+  switch (text) {
+    case 'Backspace':
+      front = textArea.value.substring(0, cursorPosition - 1);
+      back = textArea.value.substring(cursorPosition, textArea.value.length);
+      textArea.value = front + back;
+
+      // updateCursorPosition
+      cursorPosition += inputText.length;
+      textArea.selectionStart = cursorPosition - 1;
+      textArea.selectionEnd = cursorPosition - 1;
+      break;
+    case 'Del':
+      front = textArea.value.substring(0, cursorPosition);
+      back = textArea.value.substring(
+        cursorPosition + 1,
+        textArea.value.length,
+      );
+      textArea.value = front + back;
+
+      // updateCursorPosition
+      cursorPosition += inputText.length;
+      textArea.selectionStart = cursorPosition;
+      textArea.selectionEnd = cursorPosition;
+      break;
+
+    default:
+      front = textArea.value.substring(0, cursorPosition);
+      back = textArea.value.substring(cursorPosition, textArea.value.length);
+      textArea.value = front + text + back;
+
+      // updateCursorPosition
+      cursorPosition += text.length;
+      textArea.selectionStart = cursorPosition;
+      textArea.selectionEnd = cursorPosition;
+      break;
+  }
+  textArea.focus();
+}
+
+const cancelCommandKeysDefaultActions = (e) => {  
+  switch (e.key) {
+    case 'Tab':
```

Comment:
when u use switch case structure -
usually, u should have constants for case, which are declared at the top of the file
e.g.
`
const EXAMPLE = 'Example'
const NEXT = 'Next'

// your code

switch (key) {
case EXAMPLE:
//....
break;

case NEXT:
    //....
    break;

default:
    break;
}
`

Reply:
fixed